### PR TITLE
Security policy

### PR DESCRIPTION
**Pull request by v-Zer0**
These policies work on kubernetes locally, now we just need to figure out how to integrate them into the deploy script.

### CODE REVIEW

File: templates/k8s/restrict.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+spec:
+  privileged: false
+  # Required to prevent escalations to root.
+  allowPrivilegeEscalation: false
+  # This is redundant with non-root + disallow privilege escalation,
+  # but we can provide it for defense in depth.
+  requiredDropCapabilities:
+    - ALL
+  # Allow core volume types.
+  volumes:
```
Comment (zachChilders) [inline templates/k8s/restrict.yaml:19]: Do we understand these volume types?  If possible, can we restrict to 'secret'?

File: templates/k8s/restrict.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    # Assume that persistentVolumes set up by the cluster admin are safe to use.
+    - 'persistentVolumeClaim'
+  hostNetwork: false
+  hostIPC: false
+  hostPID: false
+  runAsUser:
+    # Require the container to run without root privileges.
+    rule: 'MustRunAsNonRoot'
+  seLinux:
+    # This policy assumes the nodes are using AppArmor rather than SELinux.
```
Comment (zachChilders) [inline templates/k8s/restrict.yaml:34]: apparmor?

File: deploy.ps1 [hunk 3]
```diff
@@ -180,21 +180,34 @@ $k8sReqs = @(
...
         }
     },
-    @{
-        Name     = "Harden Cluster"
-        Describe = "Apply security policy"
-        Test     = { kubectl get psp } # Improve tests
-        Set      = {
-            # Install the aks-preview extension
-            az extension add --name aks-preview
+    # @{
```
Comment (zachChilders) [inline deploy.ps1:183]: Is this supposed to be commented out?

**Review by zachChilders — CHANGES_REQUESTED** [review]

File: deploy.ps1 (hunk 3 above)
Comment (v-Zer0) [reply deploy.ps1:183]: Ya i haven't figured out how to merge it into the deploy script

File: templates/k8s/restrict.yaml (hunk 1 above)
Comment (v-Zer0) [reply templates/k8s/restrict.yaml:19]: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#volumes-and-file-systems

File: templates/k8s/restrict.yaml (hunk 2 above)
Comment (v-Zer0) [reply templates/k8s/restrict.yaml:34]: Are we using SELinux on our nodes?

**Review by zachChilders — APPROVED** [review]

**Pull request merged**